src/App.js

```diff
@@ -1,20 +1,34 @@
-import React from "react";
+import React, { Suspense } from "react";
+import NavBar from "./components/NavBar/NavBar";
 import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 
-import Home from "./pages/Home/Home";
-import Search from "./pages/Search/Search";
+const Home = React.lazy(() => import("./pages/Home/Home"));
+const Search = React.lazy(() => import("./pages/Search/Search"));
 
 import "./App.css";
+import Footer from "./components/Footer/Footer";
 
 const App = () => {
   return (
-    <Router>
-      <Switch>
-        <Route exact path="/" component={Home} />
-        <Route exact path="/search" component={Search} />
-      </Switch>
-    </Router>
+    <>
+      <Router>
+        <NavBar />
+        <Switch>
+          <Route exact path="/">
+            <Suspense fallback={<div className="loading" />}>
+              <Home />
+            </Suspense>
+          </Route>
+          <Route exact path="/search">
+            <Suspense fallback={<div className="loading" />}>
+              <Search />
+            </Suspense>
+          </Route>
+        </Switch>
+        <Footer />
+      </Router>
+    </>
```

Comment:
```suggestion
    <>
      <Router>
        <NavBar />
        <Suspense fallback={<div className="loading" />}>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/search" component={Search} />
          </Switch>
        </Suspense>
        <Footer />
      </Router>
    </>
```
Home, Search 컴포넌트를 둘 다 같은 Suspense 태그로 감쌀거면 위와 같은 방법은 어떠신가요??? 
(동작 시켜보지는 않아서 잘 동작하는지는 모르겠네요 ㅎㅎㅎ 공식문서에 있어서 한 번 제안드려봅니당 👀 https://ko.reactjs.org/docs/code-splitting.html )

Reply:
> Route render methods
The recommended method of rendering something with a <Route> is to use children elements, as shown above. There are, however, a few other methods you can use to render something with a <Route>. These are provided mostly for supporting apps that were built with earlier versions of the router before hooks were introduced. 

https://reactrouter.com/web/api/Route/route-render-methods
React Router문서에서는 children을 사용하는 것을 권장하고 있습니다~
요 부분은 component를 사용했을때 같은 페이지로 이동하게되면 컴포넌트를 unmount하고 새로 mount하는 차이가 있어 선택적으로도 사용 가능해보입니다~!

Reply:
이부분은 잘못 코멘트를 남겼네요 Suspense로 페이지 전체를 감싸는 것이 위 코드 예시에서는 더 좋겟네요